AferoSwiftSDK/Core/Models/OfflineSchedule.swift

```diff
@@ -841,14 +847,103 @@ open class OfflineSchedule: NSObject {
             public typealias Hour = UInt8
             public typealias Minute = UInt8
             
-            public var dayOfWeek: DayOfWeek
-
+            private var _daysOfWeek: Set<DayOfWeek> = []
+
+            public var useCompactDayRepresentation: Bool = false
+
+            private static let compactFlag: UInt8 = 128;
+            private static let saturdayBit: UInt8 = 1;
+            private static let fridayBit: UInt8 = 2;
+            private static let thursdayBit: UInt8 = 4;
+            private static let wednesdayBit: UInt8 = 8;
+            private static let tuesdayBit: UInt8 = 16;
+            private static let mondayBit: UInt8 = 32;
+            private static let sundayBit: UInt8 = 64;
+            
+            public var daysOfWeek: Set<DayOfWeek> {
+                get {
+                    return _daysOfWeek }
+                set {
+                    _daysOfWeek = newValue
+                    useCompactDayRepresentation = true
+                }
+            }
+            
+            public var dayOfWeek: DayOfWeek {
+                get {
+                    return daysOfWeek.first ?? DayOfWeek.sunday }
+                set { daysOfWeek = [newValue] }
+            }
+            
             @available(*, deprecated, message: "Use `dayOfWeek` instead.")
             public var localDayOfWeek: DayOfWeek {
                 get { return dayOfWeek }
                 set { dayOfWeek = newValue }
             }
             
+            
+            static func byteToDays( days: UInt8) -> Set<DayOfWeek>? {
+                var daySet: Set<DayOfWeek> = []
+                
+                if (days & TimeSpecification.compactFlag == 0) {
+                    return nil
+                }
+                
+                if (days & TimeSpecification.sundayBit != 0) {
+                    daySet.insert(DayOfWeek.sunday)
+                }
+                if (days & TimeSpecification.mondayBit != 0) {
+                    daySet.insert(DayOfWeek.monday)
+                }
+                if (days & TimeSpecification.tuesdayBit != 0) {
+                    daySet.insert(DayOfWeek.tuesday)
+                }
+                if (days & TimeSpecification.wednesdayBit != 0) {
+                    daySet.insert(DayOfWeek.wednesday)
+                }
+                if (days & TimeSpecification.thursdayBit != 0) {
+                    daySet.insert(DayOfWeek.thursday)
+                }
+                if (days & TimeSpecification.fridayBit != 0) {
+                    daySet.insert(DayOfWeek.friday)
+                }
+                if (days & TimeSpecification.saturdayBit != 0) {
+                    daySet.insert(DayOfWeek.saturday)
+                }
+                return daySet
+            }
+            
+            
+            static func daysToByte(days: Set<DayOfWeek>) ->  UInt8 {
```

Comment:
If we hang the bit indexes off of DayOfWeek as described above, the following `for` loop becomes simply:

```swift
return reduce(compactFlag) { (cur, next) -> Byte in cur | 2**next.bitIndex }
```

(or thereabouts)

Reply:
I'll try this